### PR TITLE
chore: experiments API

### PR DESCRIPTION
**Pull request by rpapani**
This PR introduces the Experiments API to list the experiments on the site. Experience Success Studio UI will call this API to get the list of experiments and metadata to render the UI

Experimentation Entity - https://github.com/adobe/spacecat-shared/pull/288

If the PR is changing the API specification:
- [x] make sure you add a "Not implemented yet" note the endpoint description, if the implementation is not ready 
  yet. Ideally, return a 501 status code with a message explaining the feature is not implemented yet.
- [x] make sure you add at least one example of the request and response.

If the PR is changing the API implementation or an entity exposed through the API:
- [x] make sure you update the API specification and the examples to reflect the changes.

## Related Issues
[SITES-22416](https://jira.corp.adobe.com/browse/SITES-22416)

### CODE REVIEW

File: docs/openapi/schemas.yaml [hunk 1]
```diff
@@ -919,6 +919,154 @@ SiteTopPageList:
...
+      type: integer
+    p_value:
+      description: The p-value of the variant for the configured metric or clicks
+      type: number
+    power:
+      description: The power of the variant for the configured metric or clicks
+      type: number
+    statsig:
+      description: The statistical significance of the variant based on the configured metric or clicks
+      type: string
```
Comment (ramboz) [inline docs/openapi/schemas.yaml:951]: Can we have an actual boolean instead?

File: docs/openapi/schemas.yaml [hunk 2]
```diff
@@ -919,6 +919,154 @@ SiteTopPageList:
...
+    views:
+      description: The number of views for the variant
+      type: integer
+    interactionsCount:
+      description: The interactions count for the variant
+      type: integer
+    p_value:
+      description: The p-value of the variant for the configured metric or clicks
+      type: number
+    power:
```
Comment (ramboz) [inline docs/openapi/schemas.yaml:946]: What does power represent from a statistical point of view?

File: docs/openapi/schemas.yaml [hunk 3]
```diff
@@ -919,6 +919,154 @@ SiteTopPageList:
...
+    url:
+      description: The url of the variant
+      $ref: '#/URL'
+    views:
+      description: The number of views for the variant
+      type: integer
+    interactionsCount:
+      description: The interactions count for the variant
+      type: integer
+    p_value:
```
Comment (ramboz) [inline docs/openapi/schemas.yaml:943]: I'm tempted to rename that to `confidence` and return `1 - p_value`

File: docs/openapi/schemas.yaml (hunk 2 above)
Comment (rpapani) [reply docs/openapi/schemas.yaml:946]: @ramboz 
I'm not really sure how we can use the power to show the information on the UI, but in the Dragos notebooks, I see we're calculating p_value, power and statsig, so I've kept all the values there.
IIRC, high power increases the confidence that a real effect exists.

File: docs/openapi/schemas.yaml (hunk 1 above)
Comment (rpapani) [reply docs/openapi/schemas.yaml:951]: In python, the true value comes as `True`, so I've kept that as string, I can convert that to boolean in js and send it as a boolean.

File: docs/openapi/schemas.yaml (hunk 3 above)
Comment (rpapani) [reply docs/openapi/schemas.yaml:943]: not sure if `(1-p_value)*100` is the confidence, may be we also have to use `power` to determine the confidence. @ddragosd any feedback on this?

**Comment by solaris007** [comment]
merge/PR should probably not be `feat` as no actual feature implementation added. rather use `chore` for docs

**Review by ekremney — APPROVED** [review]

File: docs/openapi/schemas.yaml [hunk 4]
```diff
@@ -919,6 +919,151 @@ SiteTopPageList:
...
+      type: string
+    url:
+      description: The URL of the experiment
+      $ref: '#/URL'
+    status:
+      description: The status of the experiment
+      type: string
+      enum:
+        - 'ACTIVE'
+        - 'COMPLETE'
```
Comment (ssilare-adobe) [inline docs/openapi/schemas.yaml:1005]: @rpapani Are there only two types? I assume there must be another type as well named `INACTIVE` (experiments which are paused but not completed)

**Comment by solaris007** [comment]
@rpapani i merged main branch and resolved conflicts. also node 20.15.1 is now supported.

File: docs/openapi/schemas.yaml (hunk 4 above)
Comment (rpapani) [reply docs/openapi/schemas.yaml:945]: @ramboz should we use a `DISABLED` type as well? For full experiments I've seen customers using disabled in the manifest, in that case, experiment will not run right? Do we've the `DISABLED` in experiment metadata as well, will the customer use this type or will remove the experiment metadata?

Comment (rpapani) [reply docs/openapi/schemas.yaml:945]: Looking at [experimentation documentation](https://github.com/adobe/aem-experimentation/blob/main/documentation/experiments.md), there's an `INACTIVE` status, so adding `INACTIVE` to the enum.

**Pull request merged**